src/composables/transactionList.ts

```diff
@@ -0,0 +1,338 @@
+
+const isInitalized = ref(false);
+
+const transactions = ref<ITransactionStateList>({});
+
+function generateEmptyTransactionState(): ITransactionsState {
+  return {
+    loaded: [],
+    nextPageUrl: '',
+    pending: {},
+    tipWithdrawnTransactions: [],
+  };
+}
+function getTransactionStateByAccountAddress(address: Encoded.AccountAddress) {
+  return transactions.value[address];
+}
+
+function getOrInitTransactionStateForAccount(address: Encoded.AccountAddress): ITransactionsState {
+  const accountTransactionState = getTransactionStateByAccountAddress(address);
+
+  if (accountTransactionState) {
+    return accountTransactionState;
+  }
+
+  transactions.value[address] = generateEmptyTransactionState();
+
+  return transactions.value[address];
+}
+
+function getTransactionByHash(address: Encoded.AccountAddress, hash: string) {
+  return transactions.value[address]?.loaded.find((transaction) => transaction.hash === hash);
+}
+
+function updateAccountTransactions(address: Encoded.AccountAddress, transaction: ITransaction) {
+  const transactionIndex = transactions.value[address]?.loaded?.findIndex(
+    (tr) => tr.hash === transaction.hash,
+  );
+
+  if (transactionIndex >= 0) {
+    transactions.value[address].loaded[transactionIndex] = transaction;
+  } else {
+    transactions.value[address].loaded.push(transaction);
+  }
+}
+
+const setTransactionsNextPage = (address: Encoded.AccountAddress, url: string) => {
+  transactions.value[address].nextPageUrl = url;
+};
+
+export function useTransactionList() {
+  const store = useStore();
+  const { nodeNetworkId, getAeSdk } = useAeSdk({ store });
+  const { activeAccount, isLoggedIn } = useAccounts({ store });
+
+  const allTransactionsForAccount = computed(() => {
+    if (!isLoggedIn) {
+      return [];
+    }
+    const { pending, loaded } = getOrInitTransactionStateForAccount(activeAccount.value.address);
+
+    return [...loaded, ...(pending[nodeNetworkId.value!] || [])];
+  });
+
+  function setPendingTransactionSentByHash(address: Encoded.AccountAddress, hash: string) {
+    getOrInitTransactionStateForAccount(address);
+    const index = transactions.value[address].pending[nodeNetworkId.value!]
+      ?.findIndex((transaction) => transaction.hash === hash);
+    transactions.value[address]
+      .pending[nodeNetworkId.value!][index][TX_DIRECTION.sent] = true;
+  }
+
+  function removePendingTransactionByAccount(address: Encoded.AccountAddress, hash: string) {
+    const pendingTransactionForAccount: ITransaction[] = (
+      transactions.value[address]?.pending?.[nodeNetworkId.value!] || []
+    );
+
+    if (pendingTransactionForAccount.length) {
+      transactions.value[address].pending[nodeNetworkId.value!] = (
+        pendingTransactionForAccount.filter((transaction) => transaction.hash !== hash)
+      );
+    }
+  }
+
+  async function waitTransactionMined(address: Encoded.AccountAddress, hash?: string) {
+    try {
+      if (hash) {
```

Comment:
I suggest to move this test before `try`. This way there will be no need to use exclamation mark in line 118.